website/addons/s3/views/config.py

```diff
@@ -1,79 +1,62 @@
-# -*- coding: utf-8 -*-
-
-import httplib as http
+import httplib
 
 from flask import request
-from boto.exception import BotoServerError
 
 from framework.exceptions import HTTPError
-from framework.status import push_status_message
 from framework.auth.decorators import must_be_logged_in
 
+from website.addons.s3 import utils
+from website.project.decorators import must_have_addon
 from website.project.decorators import must_have_permission
 from website.project.decorators import must_not_be_registration
-from website.project.decorators import must_have_addon
-
-from website.addons.s3.api import has_access, does_bucket_exist
-from website.addons.s3.utils import create_osf_user, serialize_urls, get_bucket_drop_down, adjust_cors  # noqa
-
-def add_s3_auth(access_key, secret_key, user_settings):
-
-    if not has_access(access_key, secret_key):
-        return False
-
-    user_name, access_key = create_osf_user(
-        access_key, secret_key, user_settings.owner.family_name
-    )
-
-    user_settings.s3_osf_user = user_name
-    user_settings.access_key = access_key['access_key_id']
-    user_settings.secret_key = access_key['secret_access_key']
-
-    user_settings.save()
-    return True
 
 
 @must_be_logged_in
 @must_have_addon('s3', 'user')
-def s3_authorize_user(user_addon, **kwargs):
-
-    s3_access_key = request.json.get('access_key')
-    s3_secret_key = request.json.get('secret_key')
-    if not s3_access_key or not s3_secret_key:
-        raise HTTPError(http.BAD_REQUEST)
-
+def s3_post_user_settings(user_addon, **kwargs):
     try:
-        if not add_s3_auth(s3_access_key, s3_secret_key, user_addon):
-            return {'message': 'Incorrect credentials'}, http.BAD_REQUEST
-    except BotoServerError:
-        #Note: Can't send back mark up :[
+        access_key = request.json['access_key']
+        secret_key = request.json['secret_key']
+    except KeyError:
+        raise HTTPError(httplib.BAD_REQUEST)
+
+    if not utils.can_list(access_key, secret_key):
         return {
-            'message': 'Could not access IAM. Please allow these keys permission.'
-        }, http.BAD_REQUEST
-    return {}
+            'message': ('Unable to list buckets.\n'
+                'Listing buckets is required permission that can be changed via IAM')
+        }, httplib.BAD_REQUEST
```

Comment:
I decided to keep it like this so I don't have to make any changes to the frontend code. All the inputs and outputs have been unchanged intentionally.